src/commands/share.ts

```diff
@@ -4,57 +4,67 @@ import readline from 'readline';
 import logger from '../logger';
 import { createShareableUrl } from '../share';
 import telemetry from '../telemetry';
-import { setupEnv } from '../util';
-import { readLatestResults } from '../util';
+import { readLatestResults, readResult, setupEnv } from '../util';
 
 export function shareCommand(program: Command) {
   program
-    .command('share')
-    .description('Create a shareable URL of your most recent eval')
+    .command('share [evalId]')
+    .description('Create a shareable URL of an eval (defaults to most recent)')
     .option('-y, --yes', 'Skip confirmation')
     .option('--env-file <path>', 'Path to .env file')
-    .action(async (cmdObj: { yes: boolean; envFile?: string } & Command) => {
-      setupEnv(cmdObj.envFile);
-      telemetry.maybeShowNotice();
-      telemetry.record('command_used', {
-        name: 'share',
-      });
-      await telemetry.send();
+    .action(
+      async (evalId: string | undefined, cmdObj: { yes: boolean; envFile?: string } & Command) => {
+        setupEnv(cmdObj.envFile);
+        telemetry.maybeShowNotice();
+        telemetry.record('command_used', {
+          name: 'share',
+        });
+        await telemetry.send();
 
-      const createPublicUrl = async () => {
-        const latestResults = await readLatestResults();
-        if (!latestResults) {
-          logger.error('Could not load results. Do you need to run `promptfoo eval` first?');
-          process.exit(1);
-        }
-        const url = await createShareableUrl(latestResults.results, latestResults.config);
-        logger.info(`View results: ${chalk.greenBright.bold(url)}`);
-      };
+        const createPublicUrl = async () => {
+          let results;
+          if (evalId) {
+            results = (await readResult(evalId))?.result;
+            if (!results) {
+              logger.error(`Could not load results for eval ID ${evalId}.`);
+              process.exit(1);
+            }
+          } else {
+            results = await readLatestResults();
+            if (!results) {
+              logger.error('Could not load results. Do you need to run `promptfoo eval` first?');
+              process.exit(1);
+            }
+          }
+          const url = await createShareableUrl(results.results, results.config);
+          logger.info(`View results: ${chalk.greenBright.bold(url)}`);
+        };
 
-      if (cmdObj.yes || process.env.PROMPTFOO_DISABLE_SHARE_WARNING) {
-        createPublicUrl();
-      } else {
-        const reader = readline.createInterface({
-          input: process.stdin,
-          output: process.stdout,
-        });
+        if (cmdObj.yes || process.env.PROMPTFOO_DISABLE_SHARE_WARNING) {
+          createPublicUrl();
+        } else {
+          const reader = readline.createInterface({
```

Comment:
nit, move the question to after where you look up the evalId. User experience is not good if you enter `promptfoo share asdasd`, confirm you want to share it, and then realize it doesn't exist.

Reply:
great point